js/ui/paygUnlockDialog.js

```diff
@@ -336,13 +411,41 @@ var PaygUnlockDialog = new Lang.Class({
         this._updateSensitivity();
     },
 
-    _showError: function(error) {
+    _processError: function(error) {
+        logError(error, 'Error adding PAYG code');
+
+        // The 'too many errors' case is a bit special, and sets a different state.
+        if (error.matches(PaygManager.PaygErrorDomain, PaygManager.PaygError.TOO_MANY_ATTEMPTS)) {
+            let currentTime = GLib.get_real_time() / GLib.USEC_PER_SEC;
+            let secondsLeft = Main.paygManager.rateLimitEndTime - currentTime;
+            let minutesLeft = Math.max(0, Math.round(secondsLeft / 60))
```

Comment:
I think this should be `Math.ceil()` rather than `Math.round()`, or the user might try again 30 seconds too early. Sorry I didn’t catch this before.

Reply:
I thought that, but if I do ceil I won't ever show the "Try in a few seconds message", since ' ceil(0.0000001) == 1`, so I thought being a little bit "imprecise" wouldn't be that bad.

Reply:
You could be safely imprecise by always adding 1 minute to `minutesLeft` if `(secondsLeft > 0)` or something like that.

Reply:
I think I'll do this:
```
            let minutesLeft = Math.max(0, Math.ceil(secondsLeft / 60))
            if (minutesLeft >= 1 || secondsLeft > 30) {
                this._setErrorMessage(Gettext.ngettext("Too many attempts. Try again in %s minute.",
                                                       "Too many attempts. Try again in %s minutes.", minutesLeft)
                                      .format(minutesLeft));
            } else {
                this._setErrorMessage(_("Too many attempts. Try again in a few seconds."));
            }
```